src/opcodes.py

```diff
@@ -0,0 +1,47 @@
+class Opcode:
+    '''This class is essentially a type to ensure that a piece of data 
+    when created is a valid operation code, 
+    or at least conforms to the format of a signed four digit base ten number'''
+
+    def __init__(self, raw):
+        self.raw = raw
+        self.sign = ""
+        self.op_list = {10:"READ", 11:"WRITE", 20:"LOAD", 21:"STORE", 30:"ADD", 31:"SUBTRACT", 32:"DIVIDE", 33:"MULTIPLY", 40:"BRANCH", 41:"BRANCHNEG", 42:"BRANCHZERO", 43:"HALT"}
+
+        if raw[0] != '-' and raw[0] != '+':
+            raise ValueError("Must have a + or - at the beginning of opcode")
+        if len(raw[1:]) != 4:
+            raise ValueError("Opcode must only have 4 integers")
+        if len(raw) == 4: ## i.e. if the opcode is unsigned, make it positive by default
```

Comment:
This won't ever be called if I'm reading that right. The if statement on line 11 will throw an exception before this code is reached